internal/apiclient/pricing.go

```diff
@@ -93,6 +118,82 @@ func NewPricingAPIClient(ctx *config.RunContext) *PricingAPIClient {
 		Currency:       currency,
 		EventsDisabled: ctx.Config.EventsDisabled,
 	}
+
+	initCache(ctx, c)
+	pricingClient = c
+	return c
+}
+
+func initCache(ctx *config.RunContext, c *PricingAPIClient) {
+	if ctx.Config.PricingCacheDisabled {
+		return
+	}
+
+	cacheFile := filepath.Join(ctx.Config.CachePath(), config.InfracostDir, "pricing.gob")
+	c.cacheFile = cacheFile
+	cache := loadCacheFromFile(cacheFile)
+	c.cache = &cache
+
+	c.mu = &sync.RWMutex{}
+	c.objectLimit = 200
+	if ctx.Config.PricingCacheObjectSize > 0 {
+		c.objectLimit = ctx.Config.PricingCacheObjectSize
+	}
+}
+
+func loadCacheFromFile(cacheFile string) map[uint64]gjson.Result {
+	cache := make(map[uint64]gjson.Result)
+	info, err := os.Stat(cacheFile)
+	if err != nil {
+		return cache
+	}
+
+	// if the cache is older than a day don't use it
+	if info.ModTime().Before(time.Now().AddDate(0, 0, -1)) {
+		return cache
+	}
+
+	f, err := os.Open(cacheFile)
+	if err != nil {
+		logging.Logger.WithError(err).Debugf("could not load cache file %s", cacheFile)
+		return cache
+	}
+
+	var storedCache map[uint64]string
+	err = gob.NewDecoder(f).Decode(&storedCache)
+	if err != nil {
+		logging.Logger.WithError(err).Debugf("failed to decode cache file %s", cacheFile)
+		return cache
+	}
+
+	for k, raw := range storedCache {
+		cache[k] = gjson.Parse(raw)
+	}
+
+	return cache
+}
+
+// FlushCache writes the in memory cache to the filesystem. This allows the cache
+// to be persisted between runs. FlushCache should only be called once, at
+// program termination.
+func (c *PricingAPIClient) FlushCache() error {
```

Comment:
Would be good to debug log the size of the objects in the cache when we write it.